### ISSUE TITLE
New open loop calibration

### ISSUE THREAD

**Issue by kals-amit** [ApolloAuto/apollo#3279]
Hi,

I am trying to generate calibration table for my vehicle. I got all the scripts to run but i have few questions:

1. Is there any documentation on how preprocess and process method in process.py works ? more specifically why this line https://github.com/ApolloAuto/apollo/blob/22367df6e6c660a0fe4ba799208b8db7aa0068c2/modules/tools/calibration/process.py#L51

2. why does speed in calibration table is upto 10 only ?

3. Is engine rpm important for calibration table generation ?

Thanks in advance.

**Comment by bbidault2** [comment]
Hello @kals-amit,

 1. The code is not documented but there is documentation that explains why and how to use these scripts here, [/docs/howto/how_to_update_vehicle_calibration.md](https://github.com/ApolloAuto/apollo/blob/r2.0.0/docs/howto/how_to_update_vehicle_calibration.md).
Regarding line `51` of `/modules/tools/calibration/process.py`, the `40` should not be hard-coded here. The goal of the function `get_start_index(data)` is to find the the line in the .csv file where the interesting part of the recording starts, when the throttle command is applied. You could replace this line by the following, 
`start_ind = np.where(data['vehicle_speed'] == 0.0)[0][0]`
Similarly, you could replace line `57` by,
`if data['vehicle_speed'][ind] == 0.0:`

2. I would say maybe because the relationship between the throttle command and the actual acceleration is linear past 10 mps, although I don't believe it. Maybe they only test their system at speeds lower than 10 mps ? 

3. Engine rpm is not important and it is never used in any of the scripts.

**Comment by kals-amit** [comment]
thanks, Boris. I forgot to update but I did that what you are saying. I just returned the first index from get_start_index as I made sure interesting data for calibration start from line 1.

for 2, I think throttle and acceleration are linearly related. I have observed this from my recordings where in you give constant throttle and you get constant acceleration. I haven't created a specific test case for this but I can and update an issue later.

for 3, you are right.
